federatedscope/vertical_fl/model/Tree.py

```diff
@@ -112,6 +114,88 @@ def update_child(self, node_num, left_child, right_child):
                   2].indicator = self.tree[node_num].indicator * right_child
 
 
+class DecisionTree(Tree):
+    def __init__(self, max_depth, lambda_, gamma):
+        super().__init__(max_depth, lambda_, gamma)
+        self.task_type = None  # ['classification', 'regression']
+
+    def _gini(self, indicator, y):
+        total_num = np.sum(indicator)
+        positive_num = np.dot(indicator, y)
+        negative_num = total_num - positive_num
+        return 1 - (positive_num / total_num)**2 - (negative_num /
+                                                    total_num)**2
+
+    def _check_same_label(self, y, indicator):
+        active_idx = np.nonzero(indicator)[0]
+        active_y = y[active_idx]
+        if np.sum(active_y) in [0, len(active_y)]:
+            return True
+        return False
+
+    def cal_gini(self, split_idx, y, indicator):
+        if self._check_same_label(y, indicator):
+            return np.float('inf')
+
+        left_child_indicator = indicator * np.concatenate(
+            [np.ones(split_idx),
+             np.zeros(len(y) - split_idx)])
+        right_child_indicator = indicator - left_child_indicator
+        left_gini = self._gini(left_child_indicator, y)
+        right_gini = self._gini(right_child_indicator, y)
+        total_num = np.sum(indicator)
+        return np.sum(left_child_indicator) / total_num * left_gini + sum(
+            right_child_indicator) / total_num * right_gini
+
+    def cal_sse(self, split_idx, y, indicator):
```

Comment:
Actually, "sse" means "Sum of Square Error of two subtrees", should we rename it?

Reply:
`sse` -> `sum_of_square_mean_err`